src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,51 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private Node<T> head;
+    private int size;
 
+    public void push(T value) {
+        head = new Node<>(value,head);
```

Comment:
You use value in `push` but `element` inside `Node`. Consider using the same term (value) everywhere for readability.
```suggestion
        head = new Node<>(value, head);
```